client.js

```diff
@@ -563,9 +537,61 @@ write-data=${ path.resolve(config.instanceDirectory, instance) }\r\n
 		confirmedOrders = [];
 		lastSignalCheck = Date.now();
 	});
-})()} else {
-	console.error("Invalid arguments, quitting.");
-	process.exit(1);
+}
+
+async function startClient() {
+	if (!fs.existsSync(config.instanceDirectory)) {
+		fs.mkdirSync(config.instanceDirectory);
+	}
+	if (!fs.existsSync("./sharedPlugins/")) {
+		fs.mkdirSync("sharedPlugins");
+	}
+	if (!fs.existsSync("./sharedMods/")) {
+		fs.mkdirSync("sharedMods");
+	}
+
+	// Set the process title, shows up as the title of the CMD window on windows
+	// and as the process name in ps/top on linux.
+	process.title = "clusterioClient "+instance;
+
+	// add better stack traces on promise rejection
+	process.on('unhandledRejection', r => console.log(r));
+
+	// make sure we have the master access token (can't write to master without it since clusterio 2.0)
+	if(!config.masterAuthToken || typeof config.masterAuthToken !== "string"){
+		console.error("ERROR invalid config!");
+		console.error("Master server now needs an access token for write operations. As clusterio slaves depends \
+		upon this, please add your token to config.json in the field named masterAuthToken. \
+		You can retrieve your auth token from the master in secret-api-token.txt after running it once.");
+	}
+
+
+	// handle commandline parameters
+	if (!command || command == "help" || command == "--help") {
```

Comment:
I think this is better handled by a switch-case and, maybe, can be extracted into a function

Reply:
For this round of refactoring I'm going to say no.  There are far better ways to do this than switches and if else trees.